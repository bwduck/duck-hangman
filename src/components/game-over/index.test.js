import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import GameOver from './index';

describe('<GameOver/>', () => {
  const msg = 'test message';

  it('renders and matches the snapshot', () => {
    const restart = () => null;
    const newGame = () => null;
    const wrapper = shallow(
      <GameOver
        gameOverMessage={msg}
        lostGame
        restart={restart}
        startNewPuzzle={newGame}
      />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('Calls restart callback when restart button is clicked', () => {
    const restart = jest.fn();
    const startNewPuzzle = jest.fn();
    const wrapper = mount(
      <GameOver
        gameOverMessage={msg}
        lostGame
        restart={restart}
        startNewPuzzle={startNewPuzzle}
      />
    );
    wrapper.find('button').simulate('click');
    expect(restart).toHaveBeenCalledTimes(1);
    expect(startNewPuzzle).toHaveBeenCalledTimes(0);
  });

  it('Calls startNewPuzzle callback when restart button is clicked', () => {
    const restart = jest.fn();
    const startNewPuzzle = jest.fn();
    const wrapper = mount(
      <GameOver
        gameOverMessage={msg}
        lostGame={false}
        restart={restart}
        startNewPuzzle={startNewPuzzle}
      />
    );
    wrapper.find('button').simulate('click');
    expect(restart).toHaveBeenCalledTimes(0);
    expect(startNewPuzzle).toHaveBeenCalledTimes(1);
  });
});
