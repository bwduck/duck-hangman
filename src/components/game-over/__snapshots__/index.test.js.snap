// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameOver/> renders and matches the snapshot 1`] = `
<div
  className="c-game-over"
>
  <span
    className="c-game-over-title"
  >
    GAME OVER!!
  </span>
  <span
    className="c-game-over-message"
  >
    test message
  </span>
  <div
    className="c-game-over-buttons"
  >
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      Try Again
    </button>
  </div>
</div>
`;
